module/VuFind/src/VuFind/Recommend/EDSDatabases.php

```diff
@@ -0,0 +1,174 @@
+ * @link     https://vufind.org/wiki/development:plugins:recommendation_modules Wiki
+ */
+class EDSDatabases implements RecommendInterface
+{
+    use \VuFind\Cache\CacheTrait;
+
+    /**
+     * Results object
+     *
+     * @var \VuFind\Search\Base\Results
+     */
+    protected $results;
+
+    /**
+     * LibGuides connector
+     *
+     * @var LibGuides
+     */
+    protected $libGuides;
+
+    /**
+     * Constructor
+     *
+     * @param LibGuides    $libGuides LibGuides API connection
+     * @param Config       $config    LibGuides API configuration object
+     * @param CacheAdapter $cache     Object cache
+     */
+    public function __construct(
+        LibGuides $libGuides,
+        Config $config,
+        CacheAdapter $cache
+    ) {
+        $this->libGuides = $libGuides;
+        $this->setCacheStorage($cache);
+
+        // Cache the data related to profiles for up to 10 minutes:
+        $this->cacheLifetime = intval($config->GetAZ->cache_lifetime ?? 600);
+    }
+
+    /**
+     * Store the configuration of the recommendation module.
+     *
+     * @param string $settings Settings from searches.ini.
+     *
+     * @return void
+     */
+    public function setConfig($settings)
+    {
+        // No action needed.
+    }
+
+    /**
+     * Called before the Search Results object performs its main search
+     * (specifically, in response to \VuFind\Search\SearchRunner::EVENT_CONFIGURED).
+     * This method is responsible for setting search parameters needed by the
+     * recommendation module and for reading any existing search parameters that may
+     * be needed.
+     *
+     * @param \VuFind\Search\Base\Params $params  Search parameter object
+     * @param \Laminas\Stdlib\Parameters $request Parameter object representing user
+     * request.
+     *
+     * @return void
+     *
+     * @SuppressWarnings(PHPMD.UnusedFormalParameter)
+     */
+    public function init($params, $request)
+    {
+        // No action needed.
+    }
+
+    /**
+     * Called after the Search Results object has performed its main search. This
+     * may be used to extract necessary information from the Search Results object
+     * or to perform completely unrelated processing.
+     *
+     * @param \VuFind\Search\Base\Results $results Search results object
+     *
+     * @return void
+     */
+    public function process($results)
+    {
+        $this->results = $results;
+    }
+
+    /**
+     * Get terms related to the query.
+     *
+     * @return array
+     */
+    public function getResults()
+    {
+        $records = $this->results->getResults();
+        $nameToDatabase = $this->getLibGuidesDatabases();
+        $databases = [];
+        foreach ($records as $record) {
+            $databaseInfo = $nameToDatabase[$record->getDbLabel()] ?? null;
+            if ($databaseInfo) {
+                $databases[$record->getDbLabel()] = $databaseInfo;
```

Comment:
It is of course ridiculous to use the actual database name/label as the match point between EDS and LibGuides AZ, but without some custom configuration there doesn't seem to be any other option.  In practice things are not always named the same, and LibGuides AZ may not be complete.  One silver lining is that LibGuides has a second "alternate titles" field that I may use as well, to see if the EDS label is a substring of it.